File: src/main/java/ac/dnd/server/admission/infrastructure/persistence/converter/StringCryptoConverter.java
package ac.dnd.server.admission.infrastructure.persistence.converter;

import org.springframework.security.crypto.encrypt.TextEncryptor;
import org.springframework.stereotype.Component;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
@Component
public class StringCryptoConverter implements AttributeConverter<String, String> {

	private final TextEncryptor textEncryptor;

	public StringCryptoConverter(TextEncryptor textEncryptor) {
		this.textEncryptor = textEncryptor;
	}

	@Override
	public String convertToDatabaseColumn(String attribute) {
		if (attribute == null) {
			return null;
		}
		try {
			return textEncryptor.encrypt(attribute);
		} catch (Exception e) {
			throw new RuntimeException(
				"Attribute encryption failed",
				e
			);
		}
	}

	@Override
	public String convertToEntityAttribute(String dbData) {
		if (dbData == null) {
			return null;
		}
		try {
			return textEncryptor.decrypt(dbData);
		} catch (Exception e) {
			throw new RuntimeException(
				"Attribute decryption failed",
				e
			);
		}
	}
}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/persistence/converter/StringCryptoConverter.kt
package ac.dnd.server.admission.infrastructure.persistence.converter

import org.springframework.security.crypto.encrypt.TextEncryptor
import org.springframework.stereotype.Component
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
@Component
class StringCryptoConverter(
    private val textEncryptor: TextEncryptor
) : AttributeConverter<String, String> {

    override fun convertToDatabaseColumn(attribute: String?): String? {
        if (attribute == null) {
            return null
        }
        return try {
            textEncryptor.encrypt(attribute)
        } catch (e: Exception) {
            throw RuntimeException("Attribute encryption failed", e)
        }
    }

    override fun convertToEntityAttribute(dbData: String?): String? {
        if (dbData == null) {
            return null
        }
        return try {
            textEncryptor.decrypt(dbData)
        } catch (e: Exception) {
            throw RuntimeException("Attribute decryption failed", e)
        }
    }
}